src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.Tests/ServiceControllerTests.cs

```diff
@@ -218,43 +220,77 @@ public void GetDevices()
 
             for (int i = 0; i < controller.DependentServices.Length; i++)
             {
-                var dependent = controller.DependentServices[i];
-                Assert.True(dependent.ServiceName.StartsWith(_testService.DependentTestServiceNamePrefix));
-                Assert.True(dependent.DisplayName.StartsWith(_testService.DependentTestServiceDisplayNamePrefix));
+                var dependent = AssertHasDependent(controller, _testService.DependentTestServiceNamePrefix + i, _testService.DependentTestServiceDisplayNamePrefix + i);
                 Assert.Equal(ServiceType.Win32OwnProcess, dependent.ServiceType);
-                Assert.Equal(0, dependent.DependentServices.Length);
+
+                // Assert that this dependent service is depended on by all the test services created after it
+                Assert.Equal(ExpectedDependentServiceCount - i - 1, dependent.DependentServices.Length);
+
+                for (int j = i + 1; j < ExpectedDependentServiceCount; j++)
+                {
+                    AssertHasDependent(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
+
+                // Assert that the dependent service depends on the main test service
+                AssertDependsOn(dependent, _testService.TestServiceName, _testService.TestServiceDisplayName);
+
+                // Assert that this dependent service depends on all the test services created before it
+                Assert.Equal(i + 1, dependent.ServicesDependedOn.Length);
+
+                for (int j = i - 1; j >= 0; j--)
+                {
+                    AssertDependsOn(dependent, _testService.DependentTestServiceNamePrefix + j, _testService.DependentTestServiceDisplayNamePrefix + j);
+                }
             }
         }
 
-        [Fact]
-        public void ServicesDependedOn()
+        public void Dispose()
         {
-            // The test service creates a number of dependent services, each of these should depend on the test service
-            var controller = new ServiceController(_testService.TestServiceName);
-            Assert.True(controller.DependentServices.Length > 0);
+            _testService.DeleteTestServices();
+        }
 
-            for (int i = 0; i < controller.DependentServices.Length; i++)
+        private static ServiceController AssertHasDependent(ServiceController controller, string serviceName, string displayName)
+        {
+            var dependent = FindService(controller.DependentServices, serviceName, displayName);
+            if (dependent == null)
             {
-                var dependent = controller.DependentServices[i];
-                Assert.True(dependent.ServicesDependedOn.Length == 1);
+                Assert.True(false, string.Format("Expected service {0} to have dependent service {1}", controller.ServiceName, serviceName));
```

Comment:
A minor argument for not providing such overloads is that you then pay the cost of the string.Format() for each message even when your tests pass.  As more tests begin to use these overloads, the more this cost becomes.

While we could add helpers like `AssertNotNull(obj, message)` ourselves, xunit actually had similar overloads which were obsolesced, the idea being that if you need custom messages for all your asserts, then your tests are probably overly complicated:
http://bradwilson.typepad.com/blog/2008/03/xunitnet-10-rc2.html
https://xunit.codeplex.com/workitem/4638

In this case, I've essentially written some custom assert overloads to provide context-specific information.  Although, I have seen some of our more data-driven tests (at least before open-sourcing) abuse `Assert.True(false, message)` though to print out test-case specific information where we could use xunit `[Theory]` to make tests more concise and get that information automatically.